### ISSUE TITLE
Fix missing IsA<_> for user callbacks

### ISSUE THREAD

**Issue by GuillaumeGomez** [gtk-rs/gir#709]
In reference to this conversation: https://github.com/gtk-rs/gtk/pull/760/files#r249967963

**Comment by EPashkin** [comment]
@sdroege That we can do automatically?
IMHO we have stable rule: it type is final - then use it directly if not - use IsA and this not depend on it parameter number.

**Comment by EPashkin** [comment]
Oh, you meant that we can't detect real type of closure.

**Comment by sdroege** [comment]
Yes, we can't know that e.g. the first parameter of the closure is the same as some parameter of the function, e.g. `&self`. So we can't know a more specific type for it other than what is provided by the .gir file.
